src/garage/sampler/on_policy_vectorized_sampler.py

```diff
@@ -15,22 +16,35 @@
 
 
 class OnPolicyVectorizedSampler(BatchSampler):
-    """BatchSampler which uses VecEnvExecutor to run multiple environments."""
+    """BatchSampler which uses VecEnvExecutor to run multiple environments.
+
+    Args:
+        algo (garage.np.algos.RLAlgorithm): An algorithm instance.
+        env (garage.envs.GarageEnv): An environement instance.
+        n_envs (int): Number of environment instances to setup.
+            This parameter has effect on sampling performance.
+
+    """
 
     def __init__(self, algo, env, n_envs=None):
         if n_envs is None:
             n_envs = singleton_pool.n_parallel * 4
         super().__init__(algo, env)
         self.n_envs = n_envs
 
+        self.vec_env = None
```

Comment:
should these go in the base class?

Reply:
I see there is some duplicate code in off- and on- policy samplers. Moving them to their base class, `BatchSampler` is a good idea. But `ISSampler` also inherits `BatchSampler`, and it does not have vec_env`. I don't feel like to make this change in this PR.